src/models/repository.js

```diff
@@ -0,0 +1,74 @@
+import timespan from "pretty-ms";
+import { getView } from "../resources/couch";
+
+const db = "tdrepo";
+const ddoc = "tdr";
+
+const getVerified = async (repo, latest = false) => {
+  let verifiedDate;
+  const firstkey = [repo];
+  const lastkey = [repo, {}];
+  const response = await getView(db, ddoc, "repoverified", {
+    limit: 1,
+    reduce: false,
+    descending: latest,
+    startkey: JSON.stringify(latest ? lastkey : firstkey),
+    endkey: JSON.stringify(latest ? firstkey : lastkey),
+  });
+  if (response.status === 200) {
+    verifiedDate = new Date(response.content.rows[0].key[1]);
+  }
+
+  return verifiedDate;
+};
+
+const getReplicate = async () => {
+  let replicate;
+
+  const response = await getView(db, ddoc, "replicate", {
+    reduce: true,
+    group_level: 1,
+  });
+  if (response.status === 200) {
+    replicate = response.content.rows;
+  }
+
+  return replicate;
+};
```

Comment:
This is returning all the data, but it isn't transforming it into an object that can be queried lower down in the map. I'd try something like:

```js
  const replicate = {};

  const response = await getView(db, ddoc, "replicate", {
    reduce: true,
    group_level: 1,
  });

  if (response.status === 200) {
    for (const row of response.content.rows) {
      replicate[row.key] = row.value;
    }
  }

  return replicate;
```

Then, lower down in the map, instead of returning the full `replicate` object for each repo, you can return

```js
replicate[repo] || 0
```

Reply:
The changes are made, Is the table form of display still good? Or do we want to change that? https://material.io/components/cards This seems to be interesting display if we are to consider changing the table format.

Reply:
Displaying these stats in table form is certainly good enough for now. The mvp.css we're using has a sort of "card" component; it's how the front page is set up. I don't know that it makes sense here, although if we end up creating a more interesting dashboard we can revisit the idea. I'm not keen on using Material Design, though, as it has a reputation in web accessibility circles of prioritizing its aesthetics over usability and inclusivity. [See here, for example.](https://www.matsuko.ca/blog/stop-using-material-design-text-fields/)